image_transport/include/image_transport/image_transport.hpp

```diff
@@ -46,16 +48,36 @@
 namespace image_transport
 {
 
+using ImagePublisher = Publisher<sensor_msgs::msg::Image>;
+
+PubLoaderPtr getPubLoader();
+SubLoaderPtr getSubLoader();
+
 /*!
  * \brief Advertise an image topic, free function version.
  */
 IMAGE_TRANSPORT_PUBLIC
-Publisher create_publisher(
+ImagePublisher::SharedPtr create_publisher(
   rclcpp::Node * node,
   const std::string & base_topic,
   rmw_qos_profile_t custom_qos = rmw_qos_profile_default,
   rclcpp::PublisherOptions options = rclcpp::PublisherOptions());
 
+ template<
+    typename MessageT,
+    typename AllocatorT = std::allocator<void>,
+    typename PublisherT = Publisher<MessageT, AllocatorT>>
+  std::shared_ptr<PublisherT>
+IMAGE_TRANSPORT_PUBLIC
+create_type_adapted_publisher(
+  rclcpp::Node * node,
```

Comment:
Switch to node interfaces like in: https://github.com/ros2/rclcpp/blob/c500695e21dd186981533bb6fb84624e4c4f752c/rclcpp/include/rclcpp/create_publisher.hpp#L46